### PR TITLE
MB-5844 Update updateMTOServiceItem to use ID from path

### PR DESCRIPTION
**Pull request by eamahanna**
## Description

This PR updates the `updatetMTOServiceItem` handler to use the ID from the request path instead of supplied in the body. It also adds validation to make sure the ID in the path and body are the same. 

This PR also updated the payload to model function to return validation errors. This will allow the endpoint to return the proper errors.  

## Setup

To run the tests, run `make server_test`

You can also test the `UpdateMTOServiceItem` endpoint using postman. 

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5844) for this change

### CODE REVIEW

File: pkg/handlers/primeapi/mto_service_item.go [hunk 1]
```diff
@@ -119,9 +119,11 @@ type UpdateMTOServiceItemHandler struct {
 func (h UpdateMTOServiceItemHandler) Handle(params mtoserviceitemops.UpdateMTOServiceItemParams) middleware.Responder {
 	logger := h.LoggerFromRequest(params.HTTPRequest)
 
-	mtoServiceItem, verrs := payloads.MTOServiceItemModelFromUpdate(params.Body)
+	mtoServiceItem, verrs := payloads.MTOServiceItemModelFromUpdate(params.MtoServiceItemID, params.Body)
+	fmt.Println(verrs)
 	if verrs != nil && verrs.HasAny() {
-		return mtoserviceitemops.NewCreateMTOServiceItemUnprocessableEntity().WithPayload(payloads.ValidationError(
+		fmt.Println("🧞‍♀️🧞‍♀️🧞‍♀️🧞‍♀️🧞‍♀️")
+		return mtoserviceitemops.NewUpdateMTOServiceItemUnprocessableEntity().WithPayload(payloads.ValidationError(
```
Comment (eamahanna) [inline pkg/handlers/primeapi/mto_service_item.go:125]: Return correct error

File: pkg/handlers/primeapi/payloads/payload_to_model.go [hunk 2]
```diff
@@ -334,7 +341,7 @@ func MTOServiceItemModelFromUpdate(mtoServiceItem primemessages.UpdateMTOService
 		if model.SITDestinationFinalAddress != nil {
 			model.SITDestinationFinalAddressID = &model.SITDestinationFinalAddress.ID
 		}
-		return model, nil
+		return model, verrs
```
Comment (eamahanna) [inline pkg/handlers/primeapi/payloads/payload_to_model.go:344]: Need to return validation errors...

File: pkg/handlers/primeapi/mto_service_item.go [hunk 3]
```diff
@@ -119,9 +119,10 @@ type UpdateMTOServiceItemHandler struct {
 func (h UpdateMTOServiceItemHandler) Handle(params mtoserviceitemops.UpdateMTOServiceItemParams) middleware.Responder {
 	logger := h.LoggerFromRequest(params.HTTPRequest)
 
-	mtoServiceItem, verrs := payloads.MTOServiceItemModelFromUpdate(params.Body)
+	mtoServiceItem, verrs := payloads.MTOServiceItemModelFromUpdate(params.MtoServiceItemID, params.Body)
+	fmt.Println(verrs)
```
Comment (LeDeep) [inline pkg/handlers/primeapi/mto_service_item.go:123]: leftover print stmt

**Review by LeDeep — APPROVED** [review]
lgtm! can be merged after removing print, see comment above.

**Pull request merged**